src/components/ShopList.tsx

```diff
@@ -42,7 +42,18 @@ const ShopList: React.FC = function () {
       return null;
     }
 
-    const filteredShops = flatten(Object.values(shops)).filter((shop) => {
+    let filteredShops: App.Shop[] = [];
+    if (selectedFilters.length === 0) {
+      filteredShops = flatten(Object.values(shops));
+    } else {
+      Object.entries(shops).forEach(([chain, chainShops]) => {
+        if (selectedFilters.includes(chain)) {
+          filteredShops.push(...chainShops);
+        }
+      });
```

Comment:
nit: use for-of loop

```suggestion
      for (const [chain, chainShops] of Object.entries(shops)) {
        if (selectedFilters.includes(chain)) {
          filteredShops.push(...chainShops);
        }
      }
```